execution_engine/src/core/runtime/mod.rs

```diff
@@ -1186,8 +1186,11 @@ where
             return Err(Error::Interpreter(error.into()).into());
         }
 
-        // For all practical purposes following cast is assumed to be safe
-        let bytes_size = key_bytes.len() as u32;
+        // For all practical purposes following conversion is assumed to be safe
+        let bytes_size: u32 = key_bytes
+            .len()
+            .try_into()
+            .expect("Keys should not serialize to many bytes");
```

Comment:
Everywhere else in this file when the conversion fails we return `Ok(Err(ApiError::OutOfMemory))`. Why not do that here as well?

Reply:
If suddenly `impl ToBytes for Key` overflowed `u32::MAX` we'd know earlier through our test suite. This is  serialization output we control, so this is justified IMO